fix(KillNemesis): report killed nemesis based on critical events

fruch opened the pull request:
since `KillNemesis` was reporting any currently running nemesis during teardown as skipped, reporting looks like it passes regardless if the teardown was cause of end of the test, of cause of critical even stopping the test.

this change would report the nemesis as failure if there are any critical events, and otherwise it would be reported as skipped.

### Testing
- [ ] - ~~https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/longevity-5gb-1h-SlaIncreaseSharesDuringLoad-aws/7/~~
- [x] - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/longevity-5gb-1h-CorruptThenRepairMonkey-aws-test/3

## PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

vponomaryov reviewed (approved):
LGTM

fruch commented:
@fgelcer I couldn't easily reproduce those cases, to validate it would works as expected 
do you have a specific job that it happens 100% ? i.e. a critical during a nemesis ?

roydahan commented:
@fruch You can just use any disruptive nemesis and set the c-s thread to CL=ALL.

fruch commented:
It's proven to be working but the error message doesn't exactly show what I had in mind, just the backtrace

fgelcer commented:
> @fgelcer I couldn't easily reproduce those cases, to validate it would works as expected do you have a specific job that it happens 100% ? i.e. a critical during a nemesis ?

`longevity-5gb-1h-GrowShrinkClusterNemesis-aws-test` after some 6 hours it always happens

fgelcer commented on test-cases/nemesis/longevity-5gb-1h-CorruptThenRepairMonkey.yaml:
```diff
@@ -2,8 +2,8 @@ test_duration: 90
 
-stress_cmd: ["cassandra-stress write cl=QUORUM duration=60m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=40 -pop seq=1..5048570 -col 'n=FIXED(8) size=FIXED(128)' -log interval=5",
-             "cassandra-stress read  cl=QUORUM duration=60m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=40 -pop seq=1..5048570 -col 'n=FIXED(8) size=FIXED(128)' -log interval=5"
+stress_cmd: ["cassandra-stress write cl=ALL duration=60m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=40 -pop seq=1..5048570 -col 'n=FIXED(8) size=FIXED(128)' -log interval=5",
```
need to revert this change

fgelcer commented on test-cases/nemesis/longevity-5gb-1h-CorruptThenRepairMonkey.yaml:
```diff
@@ -2,8 +2,8 @@ test_duration: 90
-stress_cmd: ["cassandra-stress write cl=QUORUM duration=60m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=40 -pop seq=1..5048570 -col 'n=FIXED(8) size=FIXED(128)' -log interval=5",
-             "cassandra-stress read  cl=QUORUM duration=60m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=40 -pop seq=1..5048570 -col 'n=FIXED(8) size=FIXED(128)' -log interval=5"
+stress_cmd: ["cassandra-stress write cl=ALL duration=60m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=40 -pop seq=1..5048570 -col 'n=FIXED(8) size=FIXED(128)' -log interval=5",
+             "cassandra-stress read  cl=ALL duration=60m -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)' -mode cql3 native -rate threads=40 -pop seq=1..5048570 -col 'n=FIXED(8) size=FIXED(128)' -log interval=5"
```
need to revert this change

fgelcer reviewed (approved):
other than these 2 changes were done to test it, that need to be reverted, LGTM

fruch commented:
> other than these 2 changes were done to test it, that need to be reverted, LGTM

removed the testing commit,

found one last issue, the report to ES wasn't correct and was reported as success.
fixed it and now it would be reported as failure both in ES and argus

fruch commented:
@roydahan I think the KillNemesis is proven stable, and with this fix we can backport to 2023.1, what do you say ?

roydahan commented:
For now we use it only for the nightly runs on master, so not urgent for 2023.1
Let's give it some time there and see if it's needed.

fruch commented:
> For now we use it only for the nightly runs on master, so not urgent for 2023.1
> Let's give it some time there and see if it's needed.

It was needed for the sla test cases, but not critical
